### ISSUE TITLE
Singular bugs

### ISSUE THREAD

**Issue by Mushus** [volatiletech/sqlboiler#653]
### What version of SQLBoiler are you using (`sqlboiler --version`)?

SQLBoiler v3.6.1

### What is your database and version (eg. Postgresql 10)

sqlite3

### If this happened at generation time what was the full SQLBoiler command you used to generate your models? (if not applicable leave blank)

```
sqlboiler -o ./internal/adapter/database/models sqlite3
```

### If this happened at runtime what code produced the issue? (if not applicable leave blank)

build error
```
./schema_props.go:73:9: *Schemas is not a type
./schemas.go:593:6: Schemas redeclared in this block
        previous declaration at ./schemas.go:25:6
```

### What is the output of the command above with the `-d` flag added to it? (Provided you are comfortable sharing this, it contains a blueprint of your schema)

```
using driver: /home/mushus/.gvm/pkgsets/go1.13.5/global/bin/sqlboiler-sqlite3
{"config":{"driver_name":"sqlite3","driver_config":{"blacklist":null,"dbname":"./data.db","whitelist":null},"pkg_name":"models","out_folder":"./internal/adapter/database/models","debug":true,"struct_tag_casing":"snake","imports":{"all":{"Standard":["\"database/sql\"","\"fmt\"","\"reflect\"","\"strings\"","\"sync\"","\"time\"","\"context\""],"ThirdParty":["\"github.com/friendsofgo/errors\"","\"github.com/volatiletech/sqlboiler/boil\"","\"github.com/volatiletech/sqlboiler/queries\"","\"github.com/volatiletech/sqlboiler/queries/qm\"","\"github.com/volatiletech/sqlboiler/queries/qmhelper\"","\"github.com/volatiletech/sqlboiler/strmangle\""]},"test":{"Standard":["\"bytes\"","\"reflect\"","\"testing\"","\"context\""],"ThirdParty":["\"github.com/volatiletech/sqlboiler/boil\"","\"github.com/volatiletech/sqlboiler/queries\"","\"github.com/volatiletech/sqlboiler/randomize\"","\"github.com/volatiletech/sqlboiler/strmangle\""]},"singleton":{"boil_queries":{"Standard":null,"ThirdParty":["\"github.com/volatiletech/sqlboiler/drivers\"","\"github.com/volatiletech/sqlboiler/queries\"","\"github.com/volatiletech/sqlboiler/queries/qm\""]},"boil_types":{"Standard":["\"strconv\""],"ThirdParty":["\"github.com/friendsofgo/errors\"","\"github.com/volatiletech/sqlboiler/boil\"","\"github.com/volatiletech/sqlboiler/strmangle\""]}},"test_singleton":{"boil_main_test":{"Standard":["\"database/sql\"","\"flag\"","\"fmt\"","\"math/rand\"","\"os\"","\"path/filepath\"","\"strings\"","\"testing\"","\"time\""],"ThirdParty":["\"github.com/spf13/viper\"","\"github.com/volatiletech/sqlboiler/boil\""]},"boil_queries_test":{"Standard":["\"bytes\"","\"fmt\"","\"io\"","\"io/ioutil\"","\"math/rand\"","\"regexp\""],"ThirdParty":["\"github.com/volatiletech/sqlboiler/boil\""]},"boil_suites_test":{"Standard":["\"testing\""],"ThirdParty":null},"sqlite3_main_test":{"Standard":["\"database/sql\"","\"fmt\"","\"io\"","\"math/rand\"","\"os\"","\"os/exec\"","\"path/filepath\"","\"regexp\""],"ThirdParty":["_ \"github.com/mattn/go-sqlite3\"","\"github.com/pkg/errors\"","\"github.com/spf13/viper\""]}},"based_on_type":{"null.Bool":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/null\""]},"null.Bytes":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/null\""]},"null.Float32":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/null\""]},"null.Float64":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/null\""]},"null.Int":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/null\""]},"null.Int16":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/null\""]},"null.Int32":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/null\""]},"null.Int64":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/null\""]},"null.Int8":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/null\""]},"null.String":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/null\""]},"null.Time":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/null\""]},"null.Uint":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/null\""]},"null.Uint16":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/null\""]},"null.Uint32":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/null\""]},"null.Uint64":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/null\""]},"null.Uint8":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/null\""]},"time.Time":{"Standard":["\"time\""],"ThirdParty":[]},"types.Decimal":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/sqlboiler/types\""]},"types.NullDecimal":{"Standard":[],"ThirdParty":["\"github.com/volatiletech/sqlboiler/types\""]}}},"aliases":{"tables":{"schema_prop_types":{"up_plural":"SchemaPropTypes","up_singular":"SchemaPropType","down_plural":"schemaPropTypes","down_singular":"schemaPropType","columns":{"id":"ID","key":"Key"}},"schema_props":{"up_plural":"SchemaProps","up_singular":"SchemaProp","down_plural":"schemaProps","down_singular":"schemaProp","columns":{"id":"ID","path":"Path","schema_id":"SchemaID","type_id":"TypeID"},"relationships":{"FK_0":{"local":"TypeSchemaProps","foreign":"Type"},"FK_1":{"local":"SchemaSchemaProps","foreign":"Schema"}}},"schemas":{"up_plural":"Schemas","up_singular":"Schemas","down_plural":"schemas","down_singular":"schemas","columns":{"id":"ID","name":"Name"}}}},"version":"3.6.1"},"driver_config":{"blacklist":null,"dbname":"./data.db","whitelist":null},"schema":"","dialect":{"lq":34,"rq":34,"use_index_placeholders":false,"use_last_insert_id":true,"use_schema":false,"use_default_keyword":false,"use_auto_columns":false,"use_top_clause":false,"use_output_clause":false,"use_case_when_exists_clause":false},"tables":[{"name":"schema_prop_types","schema_name":"","columns":[{"name":"id","type":"null.Int64","db_type":"INTEGER","default":"auto_increment","nullable":true,"unique":false,"validated":false,"arr_type":null,"udt_name":"","domain_name":null,"full_db_type":"INTEGER","auto_generated":false},{"name":"key","type":"string","db_type":"TEXT","default":"","nullable":false,"unique":false,"validated":false,"arr_type":null,"udt_name":"","domain_name":null,"full_db_type":"TEXT","auto_generated":false}],"p_key":{"name":"","columns":["id"]},"f_keys":null,"is_join_table":false,"to_one_relationships":null,"to_many_relationships":[{"name":"FK_0","table":"schema_prop_types","column":"id","nullable":true,"unique":false,"foreign_table":"schema_props","foreign_column":"type_id","foreign_column_nullable":true,"foreign_column_unique":false,"to_join_table":false,"join_table":"","join_local_fkey_name":"","join_local_column":"","join_local_column_nullable":false,"join_local_column_unique":false,"join_foreign_fkey_name":"","join_foreign_column":"","join_foreign_column_nullable":false,"join_foreign_column_unique":false}]},{"name":"schema_props","schema_name":"","columns":[{"name":"id","type":"null.Int64","db_type":"INTEGER","default":"auto_increment","nullable":true,"unique":false,"validated":false,"arr_type":null,"udt_name":"","domain_name":null,"full_db_type":"INTEGER","auto_generated":false},{"name":"schema_id","type":"null.Int64","db_type":"INTEGER","default":"","nullable":true,"unique":false,"validated":false,"arr_type":null,"udt_name":"","domain_name":null,"full_db_type":"INTEGER","auto_generated":false},{"name":"type_id","type":"null.Int64","db_type":"INTEGER","default":"","nullable":true,"unique":false,"validated":false,"arr_type":null,"udt_name":"","domain_name":null,"full_db_type":"INTEGER","auto_generated":false},{"name":"path","type":"string","db_type":"TEXT","default":"","nullable":false,"unique":true,"validated":false,"arr_type":null,"udt_name":"","domain_name":null,"full_db_type":"TEXT","auto_generated":false}],"p_key":{"name":"","columns":["id"]},"f_keys":[{"table":"schema_props","name":"FK_0","column":"type_id","nullable":true,"unique":false,"foreign_table":"schema_prop_types","foreign_column":"id","foreign_column_nullable":true,"foreign_column_unique":false},{"table":"schema_props","name":"FK_1","column":"schema_id","nullable":true,"unique":false,"foreign_table":"schemas","foreign_column":"id","foreign_column_nullable":true,"foreign_column_unique":false}],"is_join_table":false,"to_one_relationships":null,"to_many_relationships":null},{"name":"schemas","schema_name":"","columns":[{"name":"id","type":"null.Int64","db_type":"INTEGER","default":"auto_increment","nullable":true,"unique":false,"validated":false,"arr_type":null,"udt_name":"","domain_name":null,"full_db_type":"INTEGER","auto_generated":false},{"name":"name","type":"string","db_type":"TEXT","default":"","nullable":false,"unique":false,"validated":false,"arr_type":null,"udt_name":"","domain_name":null,"full_db_type":"TEXT","auto_generated":false}],"p_key":{"name":"","columns":["id"]},"f_keys":null,"is_join_table":false,"to_one_relationships":null,"to_many_relationships":[{"name":"FK_1","table":"schemas","column":"id","nullable":true,"unique":false,"foreign_table":"schema_props","foreign_column":"schema_id","foreign_column_nullable":true,"foreign_column_unique":false,"to_join_table":false,"join_table":"","join_local_fkey_name":"","join_local_column":"","join_local_column_nullable":false,"join_local_column_unique":false,"join_foreign_fkey_name":"","join_foreign_column":"","join_foreign_column_nullable":false,"join_foreign_column_unique":false}]}],"templates":[{"name":"templates/00_struct.go.tpl","loader":"asset:templates/00_struct.go.tpl"},{"name":"templates/01_types.go.tpl","loader":"asset:templates/01_types.go.tpl"},{"name":"templates/02_hooks.go.tpl","loader":"asset:templates/02_hooks.go.tpl"},{"name":"templates/03_finishers.go.tpl","loader":"asset:templates/03_finishers.go.tpl"},{"name":"templates/04_relationship_to_one.go.tpl","loader":"asset:templates/04_relationship_to_one.go.tpl"},{"name":"templates/05_relationship_one_to_one.go.tpl","loader":"asset:templates/05_relationship_one_to_one.go.tpl"},{"name":"templates/06_relationship_to_many.go.tpl","loader":"asset:templates/06_relationship_to_many.go.tpl"},{"name":"templates/07_relationship_to_one_eager.go.tpl","loader":"asset:templates/07_relationship_to_one_eager.go.tpl"},{"name":"templates/08_relationship_one_to_one_eager.go.tpl","loader":"asset:templates/08_relationship_one_to_one_eager.go.tpl"},{"name":"templates/09_relationship_to_many_eager.go.tpl","loader":"asset:templates/09_relationship_to_many_eager.go.tpl"},{"name":"templates/10_relationship_to_one_setops.go.tpl","loader":"asset:templates/10_relationship_to_one_setops.go.tpl"},{"name":"templates/11_relationship_one_to_one_setops.go.tpl","loader":"asset:templates/11_relationship_one_to_one_setops.go.tpl"},{"name":"templates/12_relationship_to_many_setops.go.tpl","loader":"asset:templates/12_relationship_to_many_setops.go.tpl"},{"name":"templates/13_all.go.tpl","loader":"asset:templates/13_all.go.tpl"},{"name":"templates/14_find.go.tpl","loader":"asset:templates/14_find.go.tpl"},{"name":"templates/15_insert.go.tpl","loader":"asset:templates/15_insert.go.tpl"},{"name":"templates/16_update.go.tpl","loader":"asset:templates/16_update.go.tpl"},{"name":"templates/18_delete.go.tpl","loader":"asset:templates/18_delete.go.tpl"},{"name":"templates/19_reload.go.tpl","loader":"asset:templates/19_reload.go.tpl"},{"name":"templates/20_exists.go.tpl","loader":"asset:templates/20_exists.go.tpl"},{"name":"templates/21_auto_timestamps.go.tpl","loader":"asset:templates/21_auto_timestamps.go.tpl"},{"name":"templates/singleton/boil_queries.go.tpl","loader":"asset:templates/singleton/boil_queries.go.tpl"},{"name":"templates/singleton/boil_table_names.go.tpl","loader":"asset:templates/singleton/boil_table_names.go.tpl"},{"name":"templates/singleton/boil_types.go.tpl","loader":"asset:templates/singleton/boil_types.go.tpl"},{"name":"templates_test/00_types.go.tpl","loader":"asset:templates_test/00_types.go.tpl"},{"name":"templates_test/all.go.tpl","loader":"asset:templates_test/all.go.tpl"},{"name":"templates_test/delete.go.tpl","loader":"asset:templates_test/delete.go.tpl"},{"name":"templates_test/exists.go.tpl","loader":"asset:templates_test/exists.go.tpl"},{"name":"templates_test/find.go.tpl","loader":"asset:templates_test/find.go.tpl"},{"name":"templates_test/finishers.go.tpl","loader":"asset:templates_test/finishers.go.tpl"},{"name":"templates_test/hooks.go.tpl","loader":"asset:templates_test/hooks.go.tpl"},{"name":"templates_test/insert.go.tpl","loader":"asset:templates_test/insert.go.tpl"},{"name":"templates_test/relationship_one_to_one.go.tpl","loader":"asset:templates_test/relationship_one_to_one.go.tpl"},{"name":"templates_test/relationship_one_to_one_setops.go.tpl","loader":"asset:templates_test/relationship_one_to_one_setops.go.tpl"},{"name":"templates_test/relationship_to_many.go.tpl","loader":"asset:templates_test/relationship_to_many.go.tpl"},{"name":"templates_test/relationship_to_many_setops.go.tpl","loader":"asset:templates_test/relationship_to_many_setops.go.tpl"},{"name":"templates_test/relationship_to_one.go.tpl","loader":"asset:templates_test/relationship_to_one.go.tpl"},{"name":"templates_test/relationship_to_one_setops.go.tpl","loader":"asset:templates_test/relationship_to_one_setops.go.tpl"},{"name":"templates_test/reload.go.tpl","loader":"asset:templates_test/reload.go.tpl"},{"name":"templates_test/select.go.tpl","loader":"asset:templates_test/select.go.tpl"},{"name":"templates_test/singleton/boil_main_test.go.tpl","loader":"asset:templates_test/singleton/boil_main_test.go.tpl"},{"name":"templates_test/singleton/boil_queries_test.go.tpl","loader":"asset:templates_test/singleton/boil_queries_test.go.tpl"},{"name":"templates_test/singleton/boil_suites_test.go.tpl","loader":"asset:templates_test/singleton/boil_suites_test.go.tpl"},{"name":"templates_test/singleton/sqlite3_main_test.go.tpl","loader":"base64:(sha256 of content): 78996d40b2cc529f557aaa4cf9be0e0ef498bf1cdb1d83df45c40c6236bed753"},{"name":"templates_test/types.go.tpl","loader":"asset:templates_test/types.go.tpl"},{"name":"templates_test/update.go.tpl","loader":"asset:templates_test/update.go.tpl"}]}
```

### Please provide a relevant database schema so we can replicate your issue (Provided you are comfortable sharing this)

```
CREATE TABLE IF NOT EXISTS schemas (
    id INTEGER PRIMARY KEY,
    name TEXT NOT NULL
);

CREATE TABLE IF NOT EXISTS schema_prop_types (
    id INTEGER PRIMARY KEY,
    key TEXT NOT NULL
);

CREATE TABLE IF NOT EXISTS schema_props (
    id INTEGER PRIMARY KEY,
    schema_id INTEGER,
    type_id INTEGER,
    path TEXT NOT NULL,
    foreign key (schema_id) references schemas(id),
    foreign key (type_id) references schema_prop_types(id)
);

CREATE UNIQUE INDEX IF NOT EXISTS idx_schema_props ON schema_props(path);
```

### Further information. What did you do, what did you expect?

I think this is bugs of `github.com/vattle/sqlboiler/strmangle`

https://play.golang.org/p/xjKx5wZ_XX_-

```go
// expect: schema
// actual: schemas
strmangle.Singular("schemas")
```

**Comment by aarondl** [comment]
Hi @Mushus. You can always change the name of your tables/columns with the aliases feature (see README).

I'd also accept a PR that fixes the issue.

Note that github.com/vattle/sqlboiler should not be being used anymore. vattle => volatiletech

**Comment by aarondl** [comment]
Closed by 33f7d52
